### PR TITLE
Test search_data_catalogs

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Tests for `search_data_catalogs` and its underlying functions.
* Added the possibility to search for simulations that reach a given warming level within `tas_csv`.
* Simplified how we created the main `catalog`, in `search_data_catalogs`.
* Fixed a bug when searching for alternative `experiments/members` for fixed fields.

### Does this PR introduce a breaking change?

- No

### Other information:

- The tests are based on the new catalog files in #248, so we'll either need to merge that one first, or merge this PR into 248.

### CODE REVIEW

**Review by aulemahal — APPROVED** [review]
Looks good.

File: docs/notebooks/1_catalog.ipynb [hunk 1]
```diff
@@ -254,7 +254,7 @@
     "- `allow_conversion` is used to allow searching for calculable variables, in the case where the requested variable would not be available.\n",
     "- `restrict_resolution` is used to limit the results to the finest or coarsest resolution available for each source.\n",
     "- `restrict_members` is used to limit the results to a maximum number of realizations for each source.\n",
-    "- `restrict_warming_level` is used to limit the results to only datasets that are present in the csv used for calculating warming levels.\n",
+    "- `restrict_warming_level` is used to limit the results to only datasets that are present in the csv used for calculating warming levels and/or that reach a given warming level.\n",
```
Comment (juliettelavoie) [inline docs/notebooks/1_catalog.ipynb:257]: the and/or is a bit weird here..
```suggestion
    "- `restrict_warming_level` is used to limit the results to only datasets that are present in the csv used for calculating warming levels. You can also pass a string to verify that a given warming level is reached.\n",
```

**Review by juliettelavoie — APPROVED** [review]
LGTM!

File: docs/notebooks/1_catalog.ipynb (hunk 1 above)
Comment (RondeauG) [reply docs/notebooks/1_catalog.ipynb:257]:
```suggestion
    "- `restrict_warming_level` is used to limit the results to only datasets that are present in the csv used for calculating warming levels. You can also pass a dict to verify that a given warming level is reached.\n",
```

**Pull request merged**